dev-docs/source/Architecture.rst

```diff
@@ -0,0 +1,180 @@
+one general-purpose graphical application (``workbench``) and
+many customized graphical interfaces for specific scientific techniques.
+This is illustrated, along with connections between the containers,
+in the following diagram.
+
+
+.. figure:: /images/architecture-container.png
+   :scale: 70%
+   :align: center
+   :alt: Container diagram of mantid software system
+
+   Container diagram for mantid software
+
+The ``workbench`` application is written in Python/Qt that provides access
+to the business logic for reducing/analyzing data within the ``framework``.
+Along with this it provides access to a range of visualization features
+defined within the ``mantidqt`` library - A Python/Qt library of custom
+widgets that are composed to form the ``workbench`` application.
+This library is accessible outside of the workbench application to allow
+custom applications to be built from the widgets.
+
+Please note the directional aspect of the connection/dependency arrows.
+The connections are strictly one way and those earlier in the dependency chain
+cannot contain dependencies on those later, e.g.
+
+- framework *does not* depend on ``mantidqt`` or ``workbench``
+- ``mantidqt`` *does not* depend on ``workbench``
+
+Components
+----------
+
+The following subsections describe each of the containers within the system
+in more detail.
+
+Framework
+#########
+
+The ``Framework`` refers to the data structures, algorithms
+and supporting modules that live within the ``Framework`` directory
+of the source tree - commonly referred to as "framework code".
+This part of the system is written in C++ and provides the business logic
+for reducing and analysing data independent of a graphical environment.
+The modules that compose the framework are shown within the following
+component diagram.
+
+.. figure:: /images/architecture-framework.png
+   :scale: 70%
+   :align: center
+   :alt: Component diagram of framework container
+
+   Container diagram for mantid framework
+
+The framework is separated into a base set of libraries that provide that
+core functionality e.g.
+workspace objects, instrument objects, algorithm interfaces,
+along with a set of plugin libraries that are loaded at runtime to register
+new capabilities such as algorithms for data treatment.
+
+A Python package to the C++ libraries, called ``mantid``, is defined in
+``Framework/PythonInterface`` and
+gives access to the C++-framework code from Python.
+The Python package is the primary mode of access to the C++ libraries
+for external entities.
+While the C++ API could be used directly by external parties this is
+not a supported method of interaction and API stability of the C++
+layer is not guaranteed.
+
+mantidqt
+########
+
+.. figure:: /images/architecture-mantidqt.png
+   :scale: 70%
+   :align: center
+   :alt: Component diagram of mantidqt container
+
+   Container diagram for mantidqt Python package
+
+``mantidqt`` is a Python package defined house a collcetion of Qt widgets
```

Comment:
there are typos here

Reply:
Thank you. Looking back it's clear I didn't read that part over again!